kafka/consumer.go

```diff
@@ -149,31 +151,44 @@ func NewConsumer(cfg ConsumerConfig) (*Consumer, error) {
 		cfg:      cfg,
 		client:   client,
 		consumer: consumer,
+		closed:   make(chan struct{}),
+		running:  make(chan struct{}),
 		tracer:   cfg.tracerProvider().Tracer("kafka"),
 	}, nil
 }
 
 // Close the consumer, blocking until all partition consumers are stopped.
 func (c *Consumer) Close() error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-	// Since we take the full lock when closing, there's no need to explicitly
-	// allow rebalances since polls aren't concurrent with Close().
-	c.client.Close()
-	c.consumer.wg.Wait() // Wait for all the goroutines to exit.
+	select {
+	case <-c.closed:
```

Comment:
For `c.closed` also we can use an `atomic.Bool` as for `c.running`, however, there is no panic in this case.

Reply:
Fixed in 9308bfe